File: Factory/FactoryMethod/Java/src/PizzaStore/NewYorkPizzaStore.java
package eu.svetlin.design.patterns.factory.factorymethod.PizzaStore;

import eu.svetlin.design.patterns.factory.factorymethod.Pizza.NewYorkStyle.NewYorkStyleCheesePizza;
import eu.svetlin.design.patterns.factory.factorymethod.Pizza.NewYorkStyle.NewYorkStyleClamPizza;
import eu.svetlin.design.patterns.factory.factorymethod.Pizza.NewYorkStyle.NewYorkStylePepperoniPizza;
import eu.svetlin.design.patterns.factory.factorymethod.Pizza.NewYorkStyle.NewYorkStyleVeggiePizza;
import eu.svetlin.design.patterns.factory.factorymethod.Pizza.Pizza;

public class NewYorkPizzaStore extends PizzaStore{
    @Override
    Pizza createPizza(String type) {
        if(type.equals("cheese")){
            return new NewYorkStyleCheesePizza();
        }

        if(type.equals("veggie")){
            return new NewYorkStyleVeggiePizza();
        }

        if(type.equals("clam")){
            return new NewYorkStyleClamPizza();
        }

        if(type.equals("pepperoni")){
            return new NewYorkStylePepperoniPizza();
        }

        return null;
    }
}

File: Factory/FactoryMethod/TypeScript/src/PizzaStore/NewYorkPizzaStore.ts
import NewYorkStyleCheesePizza from '../Pizza/NewYorkStyle/NewYorkStyleCheesePizza';
import NewYorkStyleClamPizza from '../Pizza/NewYorkStyle/NewYorkStyleClamPizza';
import NewYorkStylePepperoniPizza from '../Pizza/NewYorkStyle/NewYorkStylePepperoniPizza';
import NewYorkStyleVeggiePizza from '../Pizza/NewYorkStyle/NewYorkStyleVeggiePizza';
import Pizza from '../Pizza/Pizza';
import PizzaStore from './PizzaStore';

class NewYorkPizzaStore extends PizzaStore {
  createPizza(type: string): Pizza {
    if (type === 'cheese') {
      return new NewYorkStyleCheesePizza();
    }

    if (type === 'veggie') {
      return new NewYorkStyleVeggiePizza();
    }

    if (type === 'clam') {
      return new NewYorkStyleClamPizza();
    }

    if (type === 'pepperoni') {
      return new NewYorkStylePepperoniPizza();
    }

    return null;
  }
}

export default NewYorkPizzaStore;
